collector/receiver/otelarrowreceiver/internal/arrow/arrow.go

```diff
@@ -366,9 +398,11 @@ func (r *Receiver) anyStream(serverStream anyStreamServer, method string) (retEr
 			}
 		}
 
+		r.recvInFlightRequests.Add(thisCtx, 1)
 		if err := r.processAndConsume(thisCtx, method, ac, req, serverStream, authErr); err != nil {
 			return err
 		}
+		r.recvInFlightRequests.Add(thisCtx, -1)
```

Comment:
Is it intended that the recvInFlightRequests up-down counter is never decremented when processAndConsume returns an error?

Reply:
I suggest adding the increment to the top of processAndConsume, with a defer func() { decrement() } to undo that count.

Reply:
Sounds a good idea :-)

Reply:
Thanks for catching that @lquerel! This should be fixed now